packages/zero-runtime/src/styled.js

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import clsx from 'clsx';
+
+function getVariantClasses(props, variants) {
+  const variantClasses = variants
+    .filter(({ props: { __fromOwnerState = true, ...restProps } }) => {
```

Comment:
Can't we reuse the logic in https://github.com/mui/material-ui/blob/master/packages/mui-system/src/createStyled.js#L47, why do we need a new `fromOwnerState` prop?

Reply:
I did not notice this `variantsResolver`. I guess we can. I'll try to update the PR with this function and see if it works directly or if it needs some modification.